runtime/queries/sqlutil.go

```diff
@@ -48,3 +48,26 @@ func convertToDateTruncSpecifier(specifier runtimev1.TimeGrain) string {
 	}
 	panic(fmt.Errorf("unconvertable time grain specifier: %v", specifier))
 }
+
+func toTimeGrain(val string) runtimev1.TimeGrain {
+	switch strings.ToUpper(val) {
+	case "MILLISECOND":
+		return runtimev1.TimeGrain_TIME_GRAIN_MILLISECOND
+	case "SECOND":
+		return runtimev1.TimeGrain_TIME_GRAIN_SECOND
+	case "MINUTE":
+		return runtimev1.TimeGrain_TIME_GRAIN_MINUTE
+	case "HOUR":
+		return runtimev1.TimeGrain_TIME_GRAIN_HOUR
+	case "DAY":
+		return runtimev1.TimeGrain_TIME_GRAIN_DAY
+	case "WEEK":
+		return runtimev1.TimeGrain_TIME_GRAIN_WEEK
+	case "MONTH":
+		return runtimev1.TimeGrain_TIME_GRAIN_MONTH
+	case "YEAR":
+		return runtimev1.TimeGrain_TIME_GRAIN_YEAR
+	default:
+		return runtimev1.TimeGrain_TIME_GRAIN_UNSPECIFIED
+	}
```

Comment:
It's used to parse a client request parameter here `rill-developer/runtime/queries/metricsview_timeseries.go:85`

Reply:
Maybe we should parse that input separately?

- For inputs, we should return an error for invalid time grains
- For outputs, we should `panic` if not handling a time grain case

Maybe we could make the input proto definition take a `TimeGrain` enum type instead of a string? Then protobufs would do the input check automatically.